packages/mediation-manager/src/mediation-manager.ts

```diff
@@ -0,0 +1,130 @@
+import type {
+  IAgentPlugin,
+  PreMediationRequestPolicy,
+  IMediationManagerSaveMediationPolicyArgs,
+  IMediationManagerRemoveMediationPolicyArgs,
+  IMediationManagerGetMediationPolicyArgs,
+  IMediationManager,
+  IMediationGetArgs,
+  MediationResponse,
+  RequesterDid,
+  IMediationManagerSaveMediationArgs,
+  IMediationManagerRecipientDidArgs,
+  IMediationManagerAddRecipientDidArgs,
+} from '@veramo/core-types'
+import { KeyValueStore } from '@veramo/kv-store'
+
+type PreRequestPolicyStore = KeyValueStore<PreMediationRequestPolicy>
+type MediationResponseStore = KeyValueStore<MediationResponse>
+type RecipientDidStore = KeyValueStore<RequesterDid>
+
+/**
+ * Mediation Manager Plugin for {@link @veramo/core#Agent}
+ *
+ * This plugin exposes methods pertaining to the {@link @veramo/core-types#IMediationManager} interface.
+ *
+ * @remarks be advised that the {@link @veramo/core-types#IMediationManager} interface is for use with
+ * {@link @veramo/did-comm#DIDComm | DIDCOmm} and specifically the V3 Coordinate Mediation Protocol implementation.
+ *
+ * @beta This API may change without a BREAKING CHANGE notice.
+ */
+export class MediationManagerPlugin implements IAgentPlugin {
+  readonly #preRequestPolicyStore: KeyValueStore<PreMediationRequestPolicy>
+  readonly #mediationResponseStore: KeyValueStore<MediationResponse>
+  readonly #recipientDidStore: KeyValueStore<RequesterDid>
+  readonly methods: IMediationManager
+
+  constructor(
+    isMediateDefaultGrantAll = true,
+    preRequestPolicyStore: PreRequestPolicyStore,
+    mediationResponseStore: MediationResponseStore,
+    recipientDidStore: RecipientDidStore,
```

Comment:
Besides the data type of the value, can you think of situations when these kv-stores would need to be different?
Would you object to merging these into a single kv-store later on?

Reply:
I feel the trade-off here is between ease of setup and code readability. There is also intersection with the decision to modify or not modify the behavior of `kv-store` itself.
|            | agent setup | code clarity | key prefixes |
|------------|-------------|--------------|--------------|
| pre-policy | Verbose     | Good         | None         |
| response   | Verbose     | Good         | None         |
| recipients | Verbose     | Good         | None         |
| combined   | Concise     | Less Clear   | 3 prefixes   |

> [!Note]
> The context here is if `kv-store` were modified to allow `list-all` functionality. Otherwise the prefixes would just be the current `namespace` + "-list" and reference a stringified JSON value each.
> _____
> pre-policy key prefix: `pre-` -> `pre-{requester-did}`
> response key prefix: `response-` -> `response-{requester-did}`
> recipients key prefix: `recipient-` -> `recipient-{recipient-did}`

> [!Warning]
> Question 1: is the proliferation of `kv-store` namespaces (as an isolated factor) any issue?
> Question 2: is the reduced verbosity of `agent.yml` worth the complication?
> Question 3: is there a strong resistance to adding some kind of `list-all` functionality to `kv-store`? (we have multiple ways of achieving this, one that changes the underlying `keyv` and one that doesn't).

> [!Note]
> My two cents is that if we use lists of values, rather than pure `key-value` pairs, then it makes sense to combine everything into one namespace, since we will essentially just have three large rows.